drivers/overlay/overlay.go

```diff
@@ -1952,6 +1952,9 @@ func (d *Driver) ApplyDiffFromStagingDirectory(id, parent, stagingDirectory stri
 	if err := os.RemoveAll(diff); err != nil && !os.IsNotExist(err) {
 		return err
 	}
+
+	diffOutput.UncompressedDigest = diffOutput.TOCDigest
```

Comment:
The net effect of this is that
- the `Layer.UncompresssedDigest` field now has two different alternative semantics
- Which one is used is ~indeterminate (it depends on whether a partial pull was used, which is partially registry-controlled!). So the caller has no idea what the value is. And there is no record of that at all.

That might be fine, but I don’t see an _obvious_ argument. E.g. what if a blob with the same digest were both a valid-enough chunked manifest and a valid-enough tar layer?

Why is this safe?

(Extra credit: it seems natural to “just” add an alternative field and not set `Layer.UncompressedDigest` on the partial pull paths at all. But that is clearly an API break, c/image expects `UncompressedDigest` to be set. Is there a migration path that doesn’t break API without an exorbitant cost?)

Compare previous https://github.com/containers/storage/pull/1627#discussion_r1218299872 .

Reply:
This was disambiguated in https://github.com/containers/storage/pull/1719 , and c/image now understands `TOCDigest` layers where `UncompressedDigest` is unavailable.